lib/archethic_web/live/chains/node_shared_secrets_live.ex

```diff
@@ -0,0 +1,221 @@
+  end
+
+  @spec handle_event(_event :: binary(), _params :: map(), socket :: LiveView.Socket.t()) ::
+          {:noreply, LiveView.Socket.t()}
+  def handle_event(_event = "next_page", _params = %{"page" => page}, socket) do
+    {:noreply, push_patch(socket, to: Routes.live_path(socket, __MODULE__, %{"page" => page}))}
+  end
+
+  def handle_event(_event = "prev_page", _params = %{"page" => page}, socket) do
+    {:noreply, push_patch(socket, to: Routes.live_path(socket, __MODULE__, %{"page" => page}))}
+  end
+
+  @spec handle_info(
+          _msg ::
+            {:new_transaction, address :: binary(), :node_shared_secrets, DateTime.t()},
+          socket :: LiveView.Socket.t()
+        ) ::
+          {:noreply, LiveView.Socket.t()}
+  def handle_info(
+        _msg = {:new_transaction, address, :node_shared_secrets, timestamp},
+        socket = %{assigns: %{current_page: current_page, transactions: txs, tx_count: tx_count}}
+      ) do
+    display_txs = Enum.count(txs)
+
+    updated_socket =
+      case current_page do
+        1 when display_txs < @display_limit ->
+          nb_authorized_nodes = nb_of_authorized_keys(address)
+
+          socket
+          |> update(:transactions, &[display_data(address, nb_authorized_nodes, timestamp) | &1])
+          |> assign(:tx_count, tx_count + 1)
+          |> assign(:nb_authorized_nodes, nb_authorized_nodes)
+
+        1 when display_txs >= @display_limit ->
+          nb_authorized_nodes = nb_of_authorized_keys(address)
+
+          socket
+          |> assign(:tx_count, tx_count + 1)
+          |> assign(:current_page, 1)
+          |> assign(:nb_authorized_nodes, nb_authorized_nodes)
+          |> assign(:nb_pages, total_pages(tx_count + 1))
+          |> assign(:transactions, [display_data(address, nb_authorized_nodes, timestamp)])
+
+        _ ->
+          socket
+      end
+
+    {:noreply, updated_socket}
+  end
+
+  @spec transactions_from_page(current_page :: non_neg_integer(), tx_count :: non_neg_integer()) ::
+          list(map())
+  def transactions_from_page(current_page, tx_count) do
+    nb_drops = tx_count - current_page * @display_limit
+
+    {nb_drops, display_limit} =
+      if nb_drops < 0, do: {0, @display_limit + nb_drops}, else: {nb_drops, @display_limit}
+
+    case SharedSecrets.genesis_address(@txn_type) do
+      address when is_binary(address) ->
+        address
+        |> TransactionChain.list_chain_addresses()
+        |> Stream.drop(nb_drops)
+        |> Stream.take(display_limit)
+        |> Stream.map(fn {addr, timestamp} ->
+          nb_authorized_nodes = nb_of_authorized_keys(addr)
+
+          display_data(
+            addr,
+            nb_authorized_nodes,
+            DateTime.from_unix(timestamp, :millisecond) |> elem(1)
+          )
+        end)
+        |> Enum.reverse()
+
+      _ ->
+        []
+    end
+  end
+
+  @spec nb_of_authorized_keys(address :: binary()) :: non_neg_integer()
+  defp nb_of_authorized_keys(address) do
```

Comment:
I think this function can be more comprehensible. Maybe something like this can be easier to understand :

```
with {:ok, %Transaction{data: %TransactionData{ownerships: ownerships}}} <-
    TransactionChain.get_transaction(address, data: [:ownerships]),
  %Ownership{authorized_keys: authorized_keys} <- Enum.at(ownerships, 0) do
  Enum.count(authorized_keys)
else
  _ -> 0
end
```

It also handle some potential error